app/lib/frontend/static_files.dart

```diff
@@ -9,14 +9,21 @@ import 'package:mime/mime.dart' as mime;
 import 'package:path/path.dart' as path;
 
 const String _defaultStaticPath = '/static';
+const _staticRootPaths = const <String>['favicon.ico', 'robots.txt'];
 
-/// Stores binary data for /static
-final StaticsCache staticsCache = new StaticsCache();
+StaticFileCache _cache;
 
-final Set<String> staticRootFiles = new Set<String>.from([
-  '/favicon.ico',
-  '/robots.txt',
-]);
+/// The static file cache. If no cache was registered before the first access,
+/// the default instance will be created.
+StaticFileCache get staticFileCache =>
+    _cache ??= new StaticFileCache.withDefaults();
+
+/// Register the static file cache.
+/// Can be called only once, before the static file cache is set.
+void registerStaticFileCache(StaticFileCache cache) {
```

Comment:
Or void set staticFileCache? But this is fine.

Reply:
Works for me either way, but we have `registerXYZ` calls for similar stuff (although it is for scoped appengine stuff), and I'd  keep the same pattern here.

Reply:
I think it also looks more readable as a singleton